napari/layers/image/_slice.py

```diff
@@ -255,14 +262,46 @@ def _call_multi_scale(self) -> _ImageSliceResponse:
             request_id=self.id,
         )
 
-    def _slice_indices_at_level(self, level: int) -> np.ndarray:
-        indices = np.array(self.indices)
-        axes = self.dims.not_displayed
-        ds_indices = indices[axes] / self.downsample_factors[level][axes]
-        ds_indices = np.round(ds_indices.astype(float)).astype(int)
-        ds_indices = np.clip(ds_indices, 0, self.level_shapes[level][axes] - 1)
-        indices[axes] = ds_indices
-        return indices
+    def _thick_slice_at_level(self, level: int) -> _ThickNDSlice:
+        """
+        Get the data_slice rescaled for a specific level.
+        """
+        slice_arr = self.data_slice.as_array()
+        # downsample based on level
+        slice_arr /= self.downsample_factors[level]
+        slice_arr[0] = np.clip(slice_arr[0], 0, self.level_shapes[level] - 1)
+        return _ThickNDSlice.from_array(slice_arr)
+
+    def _project_thick_slice(self, data, data_slice):
+        """
+        Slice the given data with the given data slice and project the extra dims.
+        """
+        slices = []
+        slice_not_disp = data_slice[self.dims.not_displayed]
+        if self.projection_mode == 'none':
+            # early return with only the dims point being used
+            indices = np.array(slice_not_disp.point, dtype=int)
+            return data[tuple(indices)]
```

Comment:
If the data being sliced causes lazy evaluation when indexed (e.g. a dask array), then we need `asarray` (or something similar) to force it to be materialized/computed.

```suggestion
            # Ensure any lazy arrays are materialized.
            return np.asarray(data[tuple(indices)])
```

This should fix the dask test failures. Like I said, they're sometimes annoying, but sometimes make good points/catches too :)

Reply:
same thing in the return below that I suppose :+1:

Reply:
Right now, it's may not be necessary because I think the calls to `np.sum` and `np.mean` should also force evaluation, but I guess it won't do much harm either.